### DISCUSSION TITLE
Refactor the Architecture of B2K?

### DISCUSSION THREAD

**Discussion by Eneuman** [Azure/Bridge-To-Kubernetes#247] [Ideas]
@hsubramanianaks 

I have been playing around with the source code for a while now and I think I have gotten a basic understanding on how everything works. One thing that I have notices is that alot of code is in the common and libary projects even though the code is only used by one client project. From what I can tell this is because of dependency issues.

I have also notices that B2K Core functionallity code is often mixed together with implementation code (API calls etc).
This makes it hard to create unit tests and implement new functionallity.

What do you think about restructuring the code to follow Clean Architecture?

We can create a "Core" project containing Domain Entities, Service and Interfaces, and a "Infrastructure" project that contains implementations needed by the core system, ie calls to kubernetes, file system, and so on.

Todays DI setup code (that exists in common) moves to the "Client" projects.

The PR would be large but I think It would make it alot easier to understand the code for new contributors :)
What do you think?

**Comment by hsubramanianaks** [comment]
@Eneuman I would agree with this, apologies for late response. we can definitely restructure the code to make it easy to follow and easy to test. I have been working on increasing code coverage for unit tests and realized lot of the code is static and aren't individually testable. Can we discuss more about the specific details about how you are going to clean up this and then you can proceed, I can help on this as well. Please let me know. Thanks.
